core/templates/dev/head/components/entity-creation-services/exploration-creation.service.ts

```diff
@@ -88,47 +88,51 @@ angular.module('oppia').factory('ExplorationCreationService', [
         }).result.then(function(result) {
-          var yamlFile = result.yamlFile;
+          if(result!='cancel') {
+            var yamlFile = result.yamlFile;
 
-          $rootScope.loadingMessage = 'Creating exploration';
+            $rootScope.loadingMessage = 'Creating exploration';
 
-          var form = new FormData();
-          form.append('yaml_file', yamlFile);
-          form.append('payload', JSON.stringify({}));
-          CsrfTokenService.getTokenAsync().then(function(token) {
-            form.append('csrf_token', token);
-            $.ajax({
-              contentType: false,
-              data: form,
-              dataFilter: function(data) {
-                // Remove the XSSI prefix.
-                return JSON.parse(data.substring(5));
-              },
-              dataType: 'text',
-              processData: false,
-              type: 'POST',
-              url: 'contributehandler/upload'
-            }).done(function(data) {
-              $window.location = UrlInterpolationService.interpolateUrl(
-                CREATE_NEW_EXPLORATION_URL_TEMPLATE, {
-                  exploration_id: data.explorationId
-                }
-              );
-            }).fail(function(data) {
-              var transformedData = data.responseText.substring(5);
-              var parsedResponse = JSON.parse(transformedData);
-              AlertsService.addWarning(
-                parsedResponse.error || 'Error communicating with server.');
-              $rootScope.loadingMessage = '';
-              $rootScope.$apply();
+            var form = new FormData();
+            form.append('yaml_file', yamlFile);
+            form.append('payload', JSON.stringify({}));
+            CsrfTokenService.getTokenAsync().then(function(token) {
+              form.append('csrf_token', token);
+              $.ajax({
+                contentType: false,
+                data: form,
+                dataFilter: function(data) {
+                  // Remove the XSSI prefix.
+                  return JSON.parse(data.substring(5));
+                },
+                dataType: 'text',
+                processData: false,
+                type: 'POST',
+                url: 'contributehandler/upload'
+              }).done(function(data) {
+                $window.location = UrlInterpolationService.interpolateUrl(
+                  CREATE_NEW_EXPLORATION_URL_TEMPLATE, {
+                    exploration_id: data.explorationId
+                  }
+                );
+              }).fail(function(data) {
+                var transformedData = data.responseText.substring(5);
+                var parsedResponse = JSON.parse(transformedData);
+                AlertsService.addWarning(
+                  parsedResponse.error || 'Error communicating with server.');
+                $rootScope.loadingMessage = '';
+                $rootScope.$apply();
+              });
             });
-          });
+          }
+        }, function() {
+
```

Comment:
Add comments to the second arg if empty -- we've discussed that before.

Reply:
Done added in all the places Thanks!